pkg/streamingpromql/optimize/ast/prune_toggles.go

```diff
@@ -31,10 +47,34 @@ func (p *PruneToggles) Name() string {
 }
 
 func (p *PruneToggles) Apply(ctx context.Context, expr parser.Expr) (parser.Expr, error) {
-	return p.mapper.Map(ctx, expr)
+	p.mapper = NewPruneTogglesMapper()
+	p.pruneTogglesAttempts.Inc()
+	newExpr, err := p.mapper.Map(ctx, expr)
+	if p.mapper.HasChanged() {
+		p.pruneTogglesSuccesses.Inc()
+	}
+	return newExpr, err
+}
+
+func (p *PruneToggles) HasChanged() bool {
+	if p.mapper == nil {
+		return false
+	}
+	return p.mapper.HasChanged()
+}
```

Comment:
Is this method needed?

Reply:
It was only used in tests, but now we replace it by checking metrics.